schemas/layer-4.cue

```diff
@@ -1,37 +1,108 @@
 package schemas
 
 #Layer4: {
-    evaluations: [#ControlEvaluation, ...#ControlEvaluation]
+	evaluations: [#ControlEvaluation, ...#ControlEvaluation]
 }
 
-// Types
-
+// ControlEvaluation provides all assessment results for a single control
 #ControlEvaluation: {
-    name: string
-    "control-id": string
-    result: #Result
-    message: string
-    "documentation-url"?: =~"^https?://[^\\s]+$"
-    "corrupted-state"?: bool
-    "assessment-results"?: [...#AssessmentResult]
+	// Name is the name of the control being evaluated
+	name: string
+	// ControlID uniquely identifies the control
+	"control-id": string
+	// Result communicates whether the evaluation has been run, and if so, the outcome(s)
+	result: #Result
+	// Message describes the result of the evaluation
+	message: string
+	// CorruptedState is true if the control evaluation was interrupted and changes were not reverted
+	"corrupted-state"?: bool
+	// RemediationGuide provides a URL with guidance on how to remediate systems that fail control evaluation
+	"remediation-guide"?: string
+	// Assessments represents the collection of results from evaluation of each control requirement
+	assessments: [...#Assessment]
+}
+
+// Assessment provides all assessment results from evaluation of a single control requirement
+#Assessment: {
+	// RequirementID uniquely identifies the requirement being tested
+	"requirement-id": string @go(RequirementId)
+	// Applicability provides identifier strings to determine when this assessment is applicable
+	applicability: [...string]
+	// Description provides a detailed explanation of the assessment
+	description: string
+	// Result communicates whether the assessment has been run, and if so, the outcome(s)
+	result: #Result
```

Comment:
The "has been run" language here confuses me.

In what circumstances or use cases do we expect to have a document conforming to this schema that does not represent an Assessment activity that occurred in the past?

Is there value in having a layer4 payload that communicates an "intended" or "future" assessment?

Reply:
I think there are two use cases I was trying to capture:

1. Currently, steps can have the state of `NotRun` if a prior Assessment Step has failed and the process is halted.
2. Part of my intent goal with adding the `Procedures` was to allow Layer4 to be used in pre-run state (essentially like a test plan) as an input to a tool (or this library) to execute the tests and fill in the result information.